js/hotgraphicView.js

```diff
@@ -0,0 +1,230 @@
+            var selector = 'item-'+model.get('_index');
+            this.$('.hotgraphic-graphic-pin.'+selector).toggleClass('active', _isActive);
+        },
+        
+        onItemsVisitedChange: function(model, _isVisited) {
+            if (_isVisited) {
+                var selector = 'item-'+model.get('_index');
+                this.$('.hotgraphic-graphic-pin.'+selector).addClass('visited');
+            }
+        },
+
+        onPinClicked: function (event) {
+            event && event.preventDefault();
+
+            this.selectedPin = event.currentTarget;
+            var $currentHotSpot = $(event.currentTarget);
+            var currentIndex = $currentHotSpot.index() - 1;
+            $currentHotSpot.show().addClass('active');
+            
+            this.setVisited(currentIndex);
+            this.model.set('_isPopupOpen', true);
+        },
+        
+        openPopup: function(model, _isPopupOpen) {
+            if (!_isPopupOpen) {
+                Adapt.trigger('notify:close');
+                return;
+            }
+
+            this.popupView = new HotgraphicPopupView({
+                model: this.model
+            });
+
+            Adapt.trigger("notify:popup", {
+                _view: this.popupView,
+                _isCancellable: true,
+                _showCloseButton: false,
+                _closeOnBackdrop: true,
+                _classes: ' hotgraphic'
+            });
+
+            Adapt.trigger('popup:opened',  $(this.selectedPin));
+
+            this.listenTo(this.popupView, {
+                'popup:closed': this.onPopupClosed,
+                'popup:previous': this.onPopupPrevious,
+                'popup:next': this.onPopupNext
+            });
+
+            this.$('.hotgraphic-popup-inner .active').a11y_focus();
+        },
+
+        onPopupClosed: function() {
+            this.model.getActiveItem().set('_isActive', false);
+            this.removePopupEvents();
+            this.model.set('_isPopupOpen', false);
+            Adapt.trigger('popup:closed');
+        },
+
+        onPopupNext: function() {
+        },
+
+        onPopupPrevious: function() {
+        },
+
+        removePopupEvents: function() {
+            this.stopListening(this.popupView, {
+                'popup:closed': this.onPopupClosed,
+                'popup:previous': this.onPopupPrevious,
+                'popup:next': this.onPopupNext
+            });
+        },
+
+        setVisited: function(index) {
+            var item = this.model.getItem(index);
+            item.toggleActive(true);
+            item.toggleVisited(true);
+
+            var $pin = this.$('.hotgraphic-graphic-pin').eq(index);
+            // append the word 'visited.' to the pin's aria-label
+            var visitedLabel = this.model.get('_globals')._accessibility._ariaLabels.visited + ".";
+            $pin.attr('aria-label', function(index, val) {return val + " " + visitedLabel});
```

Comment:
could you pop a semi-colon in here after `}` and before `);` ?

Reply:
I guess you mean this?
```
$pin.attr('aria-label', function(index, val) {
    return val + " " + visitedLabel;
});
```

Reply:
yep - correcting my mistake, I contribute that line originally!